_layouts/post.html

```diff
@@ -30,9 +30,10 @@ <h1 class="post-title p-name" itemprop="name headline">{{ page.title | escape }}
     {{ content }}
   </div>
 
-  {%- if site.disqus.shortname -%}
-    {%- include disqus_comments.html -%}
-  {%- endif -%}
+  {% if page.comments != false and jekyll.environment == "production" %}
```

Comment:
If you get the value and it is not set on the page, it will be null and then `null != false` will be truthy.

```
---
---
{{ page.fizz | jsonify }}
```
Result is `null`

So while `page.comments != false` is unreliable, then `page.comments == true` will only be truthy if the value is true and not if it is false or missing.

```suggestion
  {% if page.comments and page.comments == true and jekyll.environment == "production" %}
```

OR

```suggestion
  {% if page.comments and page.comments == true and jekyll.environment == "production" %}
```